### PR TITLE
ACM-14308: add ocp templates

### PR DESCRIPTION
**Pull request by marek-veber**

### CODE REVIEW

File: openshift/charts/cluster-api-operator/templates/ocp-capa-provider.yaml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+data:
+  # This is used to help encode your environment variables
+  # export AWS_REGION=us-east-1 
+  # export AWS_ACCESS_KEY_ID=<your-access-key>
+  # export AWS_SECRET_ACCESS_KEY=<your-secret-access-key>
+  # export AWS_SESSION_TOKEN=<session-token> # If you are using Multi-Factor Auth.
+  # Create the base64 encoded credentials using clusterawsadm.
+  # This command uses your environment variables and encodes
+  # them in a value to be stored in a Kubernetes Secret.
+  # export AWS_B64ENCODED_CREDENTIALS=$(clusterawsadm bootstrap credentials encode-as-profile|base64 -w0)
```
Comment (serngawy) [inline openshift/charts/cluster-api-operator/templates/ocp-capa-provider.yaml:28]: I believe there is no need for the base64. check the doc ref here 
https://cluster-api-aws.sigs.k8s.io/clusterawsadm/clusterawsadm_bootstrap_credentials_encode-as-profile

File: openshift/init-cluster.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+PATH_TO_ANSICLOUD=/home/mveber/projekty/ansicloud-full/ansible-2.9
+PATH_TO_PULL_SECRETS=$PWD/pull-secrets.json
+
+
+kind create cluster
+helm install cert-manager jetstack/cert-manager --namespace cert-manager --create-namespace --set crds.enabled=true
+
+. $PATH_TO_ANSICLOUD/bin/activate
+export AWS_REGION="us-west-2"
+export AWS_ACCOUNT_ID=471112697682
```
Comment (serngawy) [inline openshift/init-cluster.sh:11]: remove account-id value, set any fake num

File: openshift/Makefile [hunk 3]
```diff
@@ -42,6 +42,7 @@ release-chart:
 	tar cf - -C $(CHART_DIR) templates/deployment.yaml templates/operator-components.yaml templates/_helpers.tpl  | tar xf - -C $(OC_CHART_DIR)
 	$(YQ) eval '.version      = $(HELM_CHART_TAG)' $(OC_CHART_YAML) -i
 	$(YQ) eval '.appVersion = "$(HELM_CHART_TAG)"' $(OC_CHART_YAML) -i
+	$(YQ) eval '.AWS_B64ENCODED_CREDENTIALS = "ZmFrZQo="' $(CHART_VALUES_FILE) -i # fake base64 string
```
Comment (serngawy) [inline openshift/Makefile:45]: please add .AWS_B64ENCODED_CREDENTIALS to values.yaml as well

File: openshift/init-cluster.sh [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+   --wait
+
+# add imagePullSecrets
+for ns in capi-operator-system capi-system capa-system; do
+kubectl create secret generic image-pull-secret \
+    --from-file=.dockerconfigjson=$PATH_TO_PULL_SECRETS \
+    --type=kubernetes.io/dockerconfigjson \
+    -n $ns
+done
+# kubectl patch serviceaccount capi-operator-manager   -p '{"imagePullSecrets": [{"name": "image-pull-secret"}]}' -n cluster-api-operator
```
Comment (serngawy) [inline openshift/init-cluster.sh:31]: remove unused code

File: openshift/init-cluster.sh [hunk 5]
```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+PATH_TO_ANSICLOUD=/home/mveber/projekty/ansicloud-full/ansible-2.9
```
Comment (serngawy) [inline openshift/init-cluster.sh:2]: why is that needed ?

**Review by serngawy — CHANGES_REQUESTED** [review]

File: openshift/init-cluster.sh (hunk 5 above)
Comment (marek-veber) [reply openshift/init-cluster.sh:2]: It is connected with:
```
. $PATH_TO_ANSICLOUD/bin/activate
export AWS_REGION="us-west-2"
export AWS_ACCOUNT_ID=471112697682
aws-saml.py --account=$AWS_ACCOUNT_ID
export AWS_PROFILE=saml
```

File: openshift/init-cluster.sh (hunk 2 above)
Comment (marek-veber) [reply openshift/init-cluster.sh]: I added there a condition checking if AWS_ACCOUNT_ID is defined externaly, ok?

**Review by serngawy — APPROVED** [review]
